CMakeLists.txt

```diff
@@ -268,6 +268,7 @@ option(IREE_BYTECODE_MODULE_ENABLE_TSAN "Enable thread sanitizer in IREE modules
 option(IREE_ENABLE_UBSAN "Enable undefined behavior sanitizer" OFF)
 option(IREE_ENABLE_SPLIT_DWARF "Enable gsplit-dwarf for debug information if the platform supports it" OFF)
 option(IREE_ENABLE_THIN_ARCHIVES "Enables thin ar archives (elf systems only). Disable for released static archives" OFF)
+option(IREE_ENABLE_COMPILER_SHARED_LIBRARY "Enables the IREE compiler shared library and uses it in all IREE tools" ON)
```

Comment:
Maybe "LINK" VS "ENABLE"? The shared library will still get built and used for plugin cases, but tools will not link against it.

Reply:
sgtm